shared/src/main/scala/crystal/viewF.scala

```diff
@@ -26,21 +24,22 @@ sealed trait ViewOps[F[_], G[_], A] {
 // The difference between a View and a StateSnapshot is that the modifier doesn't act on the current value,
 // but passes the modifier function to an external source of truth. Since we are defining no getter
 // from such source of truth, a View is defined in terms of a modifier function instead of a setter.
-final class ViewF[F[_]: Async: ContextShift, A](val get: A, val mod: (A => A) => F[Unit])
+final class ViewF[F[_]: Async, A](val get: A, val mod: (A => A) => F[Unit])
     extends ViewOps[F, Id, A] {
 
   val set: A => F[Unit] = a => mod(_ => a)
 
   // In a ViewF, we can derive modAndGet. In ViewOptF and ViewListF we have to pass it, since their
   // mod functions have no idea of the enclosing structure.
-  val modAndGet: (A => A) => F[A] = f =>
+  val modAndGet: (A => A) => F[A] = f => {
     Sync[F].delay(Promise[A]()).flatMap { p =>
```

Comment:
I tried to do this using `Deferrable` but couldn't

Reply:
Neither could I.